### ISSUE TITLE
vPools fails to create when the directory already exists in /mnt

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#626]
When looking in `/mnt`, we see this:

```
root@cmp01:~# ls /mnt
myvpool-40
```

A leftover directory managed to fail the vpool creation: (as seen in `/var/log/upstart/ovs-volumedriver-vpool-40.log`)

```
2016-04-13 11:48:30:169524 +0200 CEST -- fatal -- volumedriver_fs -- operator(): caught exception: boost::filesystem::status: Transport endpoint is not connected: "/mnt/myvpool-40" -- [0x00007f2ee07dc940]
2016-04-13 11:48:30:169601 +0200 CEST -- notice -- volumedriver_fs -- at_exit: Time spent executing user instructions (seconds): 0.484000
Time spent in operating system code (seconds): 0.632000
The maximum resident set size used, in kilobytes. 1930436
The number of page faults which were serviced without requiring any I/O: 7220
The number of page faults which were serviced by doing I/O: 0
The number of times PROCESSES was swapped entirely out of main memory: 0
The number of times the file system had to read from the disk: 0
The number of times the file system had to write to the disk: 624
The number of signals received: 0
The number of times PROCESSES voluntarily invoked a context switch: 302
The number of times an involuntary context switch took place: 169
 -- [0x00007f2ee07dc940]
```

**Comment by khenderick** [comment]
@kinvaris, @wimpers, what's the suggested behavior if the framework logic notices that it can't mount the fuse onto `/mnt/<vpool_name>`?

**Comment by wimpers** [comment]
Fail but return a meaningful error to the user. In this case I would assume it fails with /mnt/vpool name already exists.

**Comment by khenderick** [comment]
Suggested fix: `add_vpool` logic should pre-validate that the directory does not exist, the wizard should also validate this and block the wizard if the directory exists.

**Comment by wimpers** [comment]
OK.

**Comment by khenderick** [comment]
Fixed by #1038, packaged in openvstorage-2.7.4-rev.4147.6f02d53

**Comment by JeffreyDevloo** [comment]
### Information

Seeing as the GUI should now also load metadata on the vpool page, I will provide feedback whether this visual indication is working.

### Steps
- Setup new cluster
- Added disk roles
- Added backend
- Added vpool
- Created new directory 'test-check' in /mnt 
- Tried adding new vpool

### Output

Metadata being loaded in wizard (as mentioned in the information section):
![selection_013](https://cloud.githubusercontent.com/assets/17570109/19511552/f0dba3d4-95e7-11e6-965b-283fd8d883d9.png)
Feedback received by the wizard: 
![selection_014](https://cloud.githubusercontent.com/assets/17570109/19511573/0b6fb6f4-95e8-11e6-9344-e386eb96d147.png)

### Test result

Test passed.

### Packages
- openvstorage                       2.7.4-rev.4161.5f26491-1                                 amd64        openvStorage
- openvstorage-backend               1.7.4-rev.770.ae31d34-1                                  amd64        openvStorage Backend plugin
- openvstorage-backend-core          1.7.4-rev.770.ae31d34-1                                  amd64        openvStorage Backend plugin core
- openvstorage-backend-webapps       1.7.4-rev.770.ae31d34-1                                  amd64        openvStorage Backend plugin Web Applications
- openvstorage-cinder-plugin         1.2.2-rev.38.dcc3b76-1                                   amd64        OpenvStorage Cinder plugin for OpenStack
- openvstorage-core                  2.7.4-rev.4161.5f26491-1                                 amd64        openvStorage core
- openvstorage-hc                    1.7.4-rev.770.ae31d34-1                                  amd64        openvStorage Backend plugin HyperConverged
- openvstorage-sdm                   1.6.4-rev.390.89e2ad4-1                                  amd64        Open vStorage Backend ASD Manager
- openvstorage-webapps               2.7.4-rev.4161.5f26491-1                                 amd64        openvStorage Web Applications

**Comment by JeffreyDevloo** [comment]
### Information

Reopening due to shortsighted testing of the just the GUI. 
Should be tested within a flow using the Web API to see if the call also returns the error.

**Comment by khenderick** [comment]
Makes sense, I think the `add_vpool` logic might not verify the mountpoint itself. So it could fail if the GUI is bypassed.

**Comment by JeffreyDevloo** [comment]
### Information

The GUI now performs a post to check_mtpt for the chosen vpool name.
When you take this new check into account and perform it yourself, you are in the safe zone.
However if this check has not been included, it will not be executed with the add_vpool call.

### Steps
- Setup new cluster
- Added disk roles
- Added backend
- Added vpool
- Created new directory 'test-check' in /mnt 
- Tried adding new vpool via webapi

Code with check:

```
from ci.helpers.api import OVSClient
from ovs.dal.lists.storagerouterlist import StorageRouterList
from ovs.dal.lists.albabackendlist import AlbaBackendList
# Loading integration test ovsclient for password login

# Requires a backend to be already setup

class StoragerouterHelper(object):

    @staticmethod
    def get_storagerouter_by_ip(storagerouter_ip):
        """

        :param storagerouter_ip: ip of a storagerouter
        :type storagerouter_ip: str
        :return: storagerouter object
        :rtype: ovs.dal.hybrids.storagerouter.StorageRouter
        """
        return StorageRouterList.get_by_ip(storagerouter_ip)


class BackendHelper(object):

    @staticmethod
    def get_albabackend_by_name(albabackend_name):
        """
        Get a Albabackend by name

        :param albabackend_name: albabackend name
        :type albabackend_name: str
        :return: alba backend object
        :rtype: ovs.dal.hybrids.albabackend
        """

        try:
            return [alba_backend for alba_backend in AlbaBackendList.get_albabackends()
                    if alba_backend.name == albabackend_name][0]
        except IndexError:
            error_msg = "No Alba backend found with name: {0}".format(albabackend_name)
            raise NameError(error_msg)

class VPoolTest(object):

    def __init__(self):
        self.vpool_details = {
            "backend_name": "mybackend",
            "preset": "testpreset",
            "storage_ip": "10.100.199.151",
            "fragment_cache": {
                "strategy": {"cache_on_read": True, "cache_on_write": True},
                "location": "disk"
            },
            "storagedriver": {
                "sco_size": 4,
                "cluster_size": 4,
                "volume_write_buffer": 512,
                "strategy": "none",
                "global_write_buffer": 2,
                "global_read_buffer": 0,
                "deduplication": "non_dedupe",
                "dtl_transport": "tcp",
                "dtl_mode": "a_sync"
            }
        }
        self.storagerouter_ip = '10.100.199.151'
        self.vpool_name = 'test-check'
        self.api = OVSClient(self.storagerouter_ip, 'admin', 'admin')

    def validate(self, old=True):
        if old is True:
            self.test_mountpoint_old_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)
        else:
            self.test_mountpoint_new_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)

    @staticmethod
    def test_mountpoint_old_flow(storagerouter_ip, vpool_name, vpool_details, api):
        """
        Will add a vpool without checking whether or not the mount point is in use
        :return:
        """

        # Add vpool because mointpoint is not in use
        # Build ADD_VPOOL parameters
        call_parameters = {
            "call_parameters": {
                "vpool_name": vpool_name,
                "type": "alba",
                "backend_connection_info": {
                    "host": "",
                    "port": 80,
                    "username": "",
                    "password": "",
                    "backend": {
                        "backend": BackendHelper.get_albabackend_by_name(vpool_details['backend_name']).guid,
                        "metadata": vpool_details['preset']
                    }
                },
                "storage_ip": vpool_details['storage_ip'],
                "storagerouter_ip": storagerouter_ip,
                "readcache_size": int(vpool_details['storagedriver']['global_read_buffer']),
                "writecache_size": int(vpool_details['storagedriver']['global_write_buffer']),
                "fragment_cache_on_read": vpool_details['fragment_cache']['strategy']['cache_on_read'],
                "fragment_cache_on_write": vpool_details['fragment_cache']['strategy']['cache_on_write'],
                "config_params": {
                    "dtl_mode": vpool_details['storagedriver']['dtl_mode'],
                    "sco_size": int(vpool_details['storagedriver']['sco_size']),
                    "dedupe_mode": vpool_details['storagedriver']['deduplication'],
                    "cluster_size": int(vpool_details['storagedriver']['cluster_size']),
                    "write_buffer": int(vpool_details['storagedriver']['volume_write_buffer']),
                    "dtl_transport": vpool_details['storagedriver']['dtl_transport'],
                    "cache_strategy": vpool_details['storagedriver']['strategy']
                }
            }
        }

        task_guid = api.post(
            api='/storagerouters/{0}/add_vpool/'.format(
                StoragerouterHelper.get_storagerouter_by_ip(storagerouter_ip).guid),
            data=call_parameters
        )
        task_result = api.wait_for_task(task_id=task_guid, timeout=120)
        if not task_result[0]:
            error_msg = "vPool `{0}` has failed to create on storagerouter `{1}`".format(vpool_name,
                                                                                         storagerouter_ip)
            raise RuntimeError(error_msg)
        else:
            print "Creation of vPool `{0}` should have succeeded on storagerouter `{1}`".format(vpool_name,
                                                                                                storagerouter_ip)
            return storagerouter_ip, "/mnt/{0}".format(vpool_name)

    @staticmethod
    def test_mountpoint_new_flow(storagerouter_ip, vpool_name, vpool_details, api):
        """
        Will add a vpool with checking whether or not the mount point is in use
        :return:
        """
        # Check for existing mountpoint
        data = {
            'name': vpool_name
        }
        task_id = api.post(api='/storagerouters/{0}/check_mtpt/'.format(StoragerouterHelper.get_storagerouter_by_ip(storagerouter_ip).guid), data=data)
        task_result = api.wait_for_task(task_id)
        # Both are true, means that the mountpoint is not in use
        if task_result[0] is True:
            if task_result[1] is True:
                # Add vpool because mointpoint is not in use
                # Build ADD_VPOOL parameters
                call_parameters = {
                    "call_parameters": {
                        "vpool_name": vpool_name,
                        "type": "alba",
                        "backend_connection_info": {
                            "host": "",
                            "port": 80,
                            "username": "",
                            "password": "",
                            "backend": {
                                "backend": BackendHelper.get_albabackend_by_name(vpool_details['backend_name']).guid,
                                "metadata": vpool_details['preset']
                            }
                        },
                        "storage_ip": vpool_details['storage_ip'],
                        "storagerouter_ip": storagerouter_ip,
                        "readcache_size": int(vpool_details['storagedriver']['global_read_buffer']),
                        "writecache_size": int(vpool_details['storagedriver']['global_write_buffer']),
                        "fragment_cache_on_read": vpool_details['fragment_cache']['strategy']['cache_on_read'],
                        "fragment_cache_on_write": vpool_details['fragment_cache']['strategy']['cache_on_write'],
                        "config_params": {
                            "dtl_mode": vpool_details['storagedriver']['dtl_mode'],
                            "sco_size": int(vpool_details['storagedriver']['sco_size']),
                            "dedupe_mode": vpool_details['storagedriver']['deduplication'],
                            "cluster_size": int(vpool_details['storagedriver']['cluster_size']),
                            "write_buffer": int(vpool_details['storagedriver']['volume_write_buffer']),
                            "dtl_transport": vpool_details['storagedriver']['dtl_transport'],
                            "cache_strategy": vpool_details['storagedriver']['strategy']
                        }
                    }
                }

                task_guid = api.post(
                    api='/storagerouters/{0}/add_vpool/'.format(StoragerouterHelper.get_storagerouter_by_ip(storagerouter_ip).guid),
                    data=call_parameters
                )
                task_result = api.wait_for_task(task_id=task_guid, timeout=120)
                if not task_result[0]:
                    error_msg = "vPool `{0}` has failed to create on storagerouter `{1}`".format(vpool_name, storagerouter_ip)
                    raise RuntimeError(error_msg)
                else:
                    print "Creation of vPool `{0}` should have succeeded on storagerouter `{1}`".format(vpool_name, storagerouter_ip)
                    return storagerouter_ip, "/mnt/{0}".format(vpool_name)
            else:
                # Mountpoint already in use
                raise AssertionError('Mountpoint {0} already in use'.format(vpool_name))


```

### Output

**Old way of creating a vpool using API:**
The directory used was an empty one.

```
In [10]: VPoolTest().validate(old=True)
Creation of vPool `test-check` should have succeeded on storagerouter `10.100.199.151`
```

**New way with the check**:
The directory used was an empty one:

```
In [11]: VPoolTest().validate(old=False)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-11-a2fcc41bb374> in <module>()
----> 1 VPoolTest().validate(old=False)

<ipython-input-9-d621c2d42d15> in validate(self, old)
     71             self.test_mountpoint_old_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)
     72         else:
---> 73             self.test_mountpoint_new_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)
     74 
     75     @staticmethod

<ipython-input-9-d621c2d42d15> in test_mountpoint_new_flow(storagerouter_ip, vpool_name, vpool_details, api)
    191             else:
    192                 # Mountpoint already in use
--> 193                 raise AssertionError('Mountpoint {0} already in use'.format(vpool_name))

AssertionError: Mountpoint test-check already in use
```

### Test result

The vpool creation did not error even though the mountpoint already existed, the volumedriver will validate whether the directory is empty or not (see test below).
The old way (without checking the mtpt) did not fail in my case.
Test passed.

**Comment by JeffreyDevloo** [comment]
### Information

To test my conclusion I conducted another test: adding a file to the directory.

### Steps
- Setup new cluster
- Added disk roles
- Added backend
- Added vpool
- Created new directory 'test-check' in /mnt 
- Touched a file 'test' in the directory
- Tried adding new vpool via webapi (old way)

The code used is the identical the code posted above.

### Output

```
In [6]: vpooltest.validate(old=True)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-6-8a10ab9406e8> in <module>()
----> 1 vpooltest.validate(old=True)

<ipython-input-1-d621c2d42d15> in validate(self, old)
     69     def validate(self, old=True):
     70         if old is True:
---> 71             self.test_mountpoint_old_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)
     72         else:
     73             self.test_mountpoint_new_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)

<ipython-input-1-d621c2d42d15> in test_mountpoint_old_flow(storagerouter_ip, vpool_name, vpool_details, api)
    123             error_msg = "vPool `{0}` has failed to create on storagerouter `{1}`".format(vpool_name,
    124                                                                                          storagerouter_ip)
--> 125             raise RuntimeError(error_msg)
    126         else:
    127             print "Creation of vPool `{0}` should have succeeded on storagerouter `{1}`".format(vpool_name,

RuntimeError: vPool `test-check` has failed to create on storagerouter `10.100.199.151`

```

Volumedriver logging:

```
2016-10-19 12:39:54 743924 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VFSMetaDataStore - 00000000000000ec - info - maybe_initialise_: Filesystem metadata store already initialized
2016-10-19 12:39:54 744088 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/PeriodicAction - 00000000000000ed - info - PeriodicAction: Started periodic action StatsCollector, period: 6000 s
2016-10-19 12:39:54 744181 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VFSStatsCollector - 00000000000000ee - info - StatsCollector: up and running
2016-10-19 12:39:54 744235 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VFSStatsCollectorComponent - 00000000000000ef - info - StatsCollectorComponent: up and running
2016-10-19 12:39:54 744283 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/MTServer - 00000000000000f0 - info - MTServer: MT SERVER
2016-10-19 12:39:54 744340 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/MTServer - 00000000000000f1 - info - bindAndListen: Port 26206 Backlog 32
2016-10-19 12:39:54 744411 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/MTServer - 00000000000000f2 - info - bindAndListen: MTServer listening on port 26206, fd 39
2016-10-19 12:39:54 746087 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FileSystem - 00000000000000f3 - info - FileSystem: 0831830d-3d71-4697-a9d8-f937b8be2c1d initialized
2016-10-19 12:39:54 746240 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000f4 - info - operator(): 0831830d-3d71-4697-a9d8-f937b8be2c1d: mounting at "/mnt/test-check"
2016-10-19 12:39:54 746318 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000f5 - info - operator(): 1: -f
2016-10-19 12:39:54 746354 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000f6 - info - operator(): 2: -o
2016-10-19 12:39:54 746380 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000f7 - info - operator(): 3: big_writes
2016-10-19 12:39:54 746407 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000f8 - info - operator(): 4: -o
2016-10-19 12:39:54 746433 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000f9 - info - operator(): 5: sync_read
2016-10-19 12:39:54 746469 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000fa - info - operator(): 6: -o
2016-10-19 12:39:54 746493 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000fb - info - operator(): 7: allow_other
2016-10-19 12:39:54 746517 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000fc - info - operator(): 8: -o
2016-10-19 12:39:54 746550 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000fd - info - operator(): 9: use_ino
2016-10-19 12:39:54 746572 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000fe - info - operator(): 10: -o
2016-10-19 12:39:54 746591 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 00000000000000ff - info - operator(): 11: default_permissions
2016-10-19 12:39:54 746611 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 0000000000000100 - info - operator(): 12: -o
2016-10-19 12:39:54 746631 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 0000000000000101 - info - operator(): 13: uid=1001
2016-10-19 12:39:54 746649 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 0000000000000102 - info - operator(): 14: -o
2016-10-19 12:39:54 746668 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 0000000000000103 - info - operator(): 15: gid=1001
2016-10-19 12:39:54 746688 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 0000000000000104 - info - operator(): 16: -o
2016-10-19 12:39:54 746707 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 0000000000000105 - info - operator(): 17: umask=0002
fuse: mountpoint is not empty
fuse: if you are sure this is safe, use the 'nonempty' mount option
2016-10-19 12:39:54 746860 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 0000000000000106 - error - operator(): 0831830d-3d71-4697-a9d8-f937b8be2c1d: fuse_setup_common failed
2016-10-19 12:39:55 747241 +0200 - ovs-node1 - 24292/0x00007fd8f2ffd700 - volumedriverfs/MTServer - 0000000000000107 - info - exit: Exiting the work procedure 
2016-10-19 12:39:55 747356 +0200 - ovs-node1 - 24292/0x00007fd8f2ffd700 - volumedriverfs/MTServer - 0000000000000108 - info - shutdown: Shutting down the server
2016-10-19 12:39:55 747388 +0200 - ovs-node1 - 24292/0x00007fd8f2ffd700 - volumedriverfs/MTServer - 0000000000000109 - info - close: Closing fd 39
2016-10-19 12:39:55 747445 +0200 - ovs-node1 - 24292/0x00007fd8f2ffd700 - volumedriverfs/MTServer - 000000000000010a - info - cleanup_threads_locked: Trying to join 0 threads
2016-10-19 12:39:55 747660 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FileSystem - 000000000000010b - info - ~FileSystem: 0831830d-3d71-4697-a9d8-f937b8be2c1d: sayonara
2016-10-19 12:39:55 747816 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/MTServer - 000000000000010c - info - shutdown: Shutting down the server
2016-10-19 12:39:55 747842 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/MTServer - 000000000000010d - info - close: Not closing fd -1
2016-10-19 12:39:55 747859 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/MTServer - 000000000000010e - info - cleanup_threads_locked: Trying to join 0 threads
2016-10-19 12:39:55 748047 +0200 - ovs-node1 - 24292/0x00007fd8f37fe700 - volumedriverfs/PeriodicAction - 000000000000010f - info - run: StatsCollector: exiting
2016-10-19 12:39:55 748216 +0200 - ovs-node1 - 24292/0x00007fd8fb7fe700 - volumedriverfs/PeriodicAction - 0000000000000110 - info - run: ScrubManager: exiting
2016-10-19 12:39:55 748305 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VolManager - 0000000000000111 - info - stop: Stopping the Volume Manager: should only happen on shutdown
2016-10-19 12:39:55 748341 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VolManager - 0000000000000112 - info - stop: Cleaning up 14 components.
2016-10-19 12:39:55 748360 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VolManager - 0000000000000113 - info - ~VolManager: stopping periodic actions
2016-10-19 12:39:55 748443 +0200 - ovs-node1 - 24292/0x00007fd902330700 - volumedriverfs/PeriodicAction - 0000000000000114 - info - run: SCOCacheCleaner: exiting
2016-10-19 12:39:55 748603 +0200 - ovs-node1 - 24292/0x00007fd901b2f700 - volumedriverfs/PeriodicAction - 0000000000000115 - info - run: SCOCacheAccessDataPersistor: exiting
2016-10-19 12:39:55 748840 +0200 - ovs-node1 - 24292/0x00007fd90132e700 - volumedriverfs/PeriodicAction - 0000000000000116 - info - run: MetaDataFreeSpaceChecker: exiting
2016-10-19 12:39:55 749099 +0200 - ovs-node1 - 24292/0x00007fd900b2d700 - volumedriverfs/PeriodicAction - 0000000000000117 - info - run: FailOverCacheChecker: exiting
2016-10-19 12:39:55 749257 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VolManager - 0000000000000118 - info - ~VolManager: stopping thread pool
2016-10-19 12:39:55 750044 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VolManager - 0000000000000119 - info - ~VolManager: Exiting volumedriver destructor
2016-10-19 12:39:55 750116 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/LocORemServer - 000000000000011a - info - stop_: 26301: stopping
2016-10-19 12:39:55 750195 +0200 - ovs-node1 - 24292/0x00007fd902b31700 - volumedriverfs/LocORemServer - 000000000000011b - info - work_: I/O service exited, breaking out of loop
2016-10-19 12:39:55 750342 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/LocORemServer - 000000000000011c - info - stop_: 26301: done
2016-10-19 12:39:55 750411 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/MetaDataServerDataBase - 000000000000011d - info - ~DataBase: Releasing all open tables (0)
2016-10-19 12:39:55 750434 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/MetaDataServerRocksDataBase - 000000000000011e - info - ~RocksDataBase: Releasing all tables (0)
2016-10-19 12:39:55 751074 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/PeriodicActionPool - 000000000000011f - info - stop_: mds-periodic-actions: stopping
2016-10-19 12:39:55 751180 +0200 - ovs-node1 - 24292/0x00007fd913fff700 - volumedriverfs/PeriodicActionPool - 0000000000000120 - info - run_: mds-periodic-actions: I/O service exited
2016-10-19 12:39:55 751572 +0200 - ovs-node1 - 24292/0x00007fd9137fe700 - volumedriverfs/PeriodicActionPool - 0000000000000121 - info - run_: mds-periodic-actions: I/O service exited
2016-10-19 12:39:55 751696 +0200 - ovs-node1 - 24292/0x00007fd912ffd700 - volumedriverfs/PeriodicActionPool - 0000000000000122 - info - run_: mds-periodic-actions: I/O service exited
2016-10-19 12:39:55 751802 +0200 - ovs-node1 - 24292/0x00007fd9127fc700 - volumedriverfs/PeriodicActionPool - 0000000000000123 - info - run_: mds-periodic-actions: I/O service exited
2016-10-19 12:39:55 752577 +0200 - ovs-node1 - 24292/0x00007fd8fbfff700 - volumedriverfs/PeriodicAction - 0000000000000124 - info - run: VolumeLockReaper: exiting

2016-10-19 12:39:55 753245 +0200 - ovs-node1 - 24292/0x00007fd8faffd700 - volumedriverfs/ZWorkerPool - 0000000000000125 - info - route_: ObjectRouterWorkerPool: stop requested, breaking out of the loop
2016-10-19 12:39:55 754294 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/VFSObjectRouter - 0000000000000126 - info - ~ObjectRouter: Bye
2016-10-19 12:39:55 754425 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/EventPublisher - 0000000000000127 - info - ~EventPublisher: Stopping
2016-10-19 12:39:55 755509 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/volumedriver_fs - 0000000000000128 - fatal - operator(): caught exception: problem running filesystem 
2016-10-19 12:39:55 756893 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/volumedriver_fs - 0000000000000129 - notice - at_exit: Time spent executing user instructions (seconds): 0.36000
Time spent in operating system code (seconds): 0.32000
The maximum resident set size used, in kilobytes. 33892
The number of page faults which were serviced without requiring any I/O: 1777
The number of page faults which were serviced by doing I/O: 0
The number of times PROCESSES was swapped entirely out of main memory: 0
The number of times the file system had to read from the disk: 0
The number of times the file system had to write to the disk: 40
The number of signals received: 0
The number of times PROCESSES voluntarily invoked a context switch: 212
The number of times an involuntary context switch took place: 243
```

TLDR volumedriver logging: 

```
fuse: mountpoint is not empty
fuse: if you are sure this is safe, use the 'nonempty' mount option
2016-10-19 12:39:54 746860 +0200 - ovs-node1 - 24292/0x00007fd9371d7940 - volumedriverfs/FuseInterface - 0000000000000106 - error - operator(): 0831830d-3d71-4697-a9d8-f937b8be2c1d: fuse_setup_common failed
```

### Test result

The creation of the vpool will fail if the directory is not empty.
OVS' web api will not report this back to the user nor will it validate whether the vpool is allowed to be created (to prevent the volumedriver issue)
Test failed.

### Required change

add_vpool call of the web api should check whether or not the mount point is in use.

**Comment by JeffreyDevloo** [comment]
### Information

Actually closing this ticket as it assumed that the directory was at fault.
With further testing we have pinpointed that files within the leftover directory were the true culprit.

(see reference ticket)
